cli/src/commands/rebase.rs

```diff
@@ -451,164 +453,74 @@ fn rebase_descendants_transaction(
-    let new_children: Vec<_> = new_children_expression
-        .evaluate_programmatic(workspace_command.repo().as_ref())?
-        .iter()
-        .commits(workspace_command.repo().store())
-        .try_collect()?;
-    workspace_command.check_rewritable(new_children.iter().ids())?;
-
-    move_commits_transaction(
-        ui,
-        settings,
-        workspace_command,
-        &new_parent_ids,
-        &new_children,
-        target_commits,
-        None,
-        rebase_options,
-    )
-}
-
-fn rebase_revisions_before(
-    ui: &mut Ui,
-    settings: &UserSettings,
-    workspace_command: &mut WorkspaceCommandHelper,
-    before_commits: &IndexSet<Commit>,
-    target_commits: &[Commit],
-    rebase_options: &RebaseOptions,
-) -> Result<(), CommandError> {
-    workspace_command.check_rewritable(target_commits.iter().ids())?;
-    let before_commit_ids = before_commits.iter().ids().cloned().collect_vec();
-    workspace_command.check_rewritable(&before_commit_ids)?;
-
-    let new_children_expression = RevsetExpression::commits(before_commit_ids);
-    let new_parents_expression = new_children_expression.parents();
-
-    ensure_no_commit_loop(
-        workspace_command.repo().as_ref(),
-        &new_children_expression,
-        &new_parents_expression,
-    )?;
-
-    // Not using `new_parents_expression` here to persist the order of parents
-    // specified in `before_commits`.
-    let new_parent_ids: IndexSet<_> = before_commits
-        .iter()
-        .flat_map(|commit| commit.parent_ids().iter().cloned().collect_vec())
-        .collect();
-    let new_parent_ids = new_parent_ids.into_iter().collect_vec();
-    let new_children = before_commits.iter().cloned().collect_vec();
-
-    move_commits_transaction(
-        ui,
-        settings,
-        workspace_command,
-        &new_parent_ids,
-        &new_children,
-        target_commits,
-        None,
-        rebase_options,
-    )
-}
-
-#[allow(clippy::too_many_arguments)]
-fn rebase_revisions_after_before(
-    ui: &mut Ui,
-    settings: &UserSettings,
-    workspace_command: &mut WorkspaceCommandHelper,
-    after_commits: &IndexSet<Commit>,
-    before_commits: &IndexSet<Commit>,
-    target_commits: &[Commit],
-    rebase_options: &RebaseOptions,
-) -> Result<(), CommandError> {
-    workspace_command.check_rewritable(target_commits.iter().ids())?;
-    let before_commit_ids = before_commits.iter().ids().cloned().collect_vec();
-    workspace_command.check_rewritable(&before_commit_ids)?;
-
-    let after_commit_ids = after_commits.iter().ids().cloned().collect_vec();
-    let new_children_expression = RevsetExpression::commits(before_commit_ids);
-    let new_parents_expression = RevsetExpression::commits(after_commit_ids.clone());
+    };
+    let destination_commits = resolve_revisions(destination)?;
+    let after_commits = resolve_revisions(insert_after)?;
+    let before_commits = resolve_revisions(insert_before)?;
```

Comment:
nit: (as a follow up) these args can be changed to `Option<Vec<Revisionarg>>` type. Then, `.is_empty()` can be replaced with match of `Some(_)`s.